pkg/oci/load_balancer.go

```diff
@@ -230,7 +249,8 @@ func (cp *CloudProvider) ensureSSLCertificate(ctx context.Context, lb *loadbalan
 	}
 
 	// Although we iterate here only one certificate is supported at the moment.
-	certs, err := spec.Certificates()
+	certs := make(map[string]loadbalancer.CertificateDetails)
```

Comment:
This is a strange pattern. Any reason you're not allocating the map in the function and returning?

Reply:
It's because I wanted to hold all certificates (both for listeners and backend sets) within the same map. Therefore, I am creating a map of certificates and I add to it step by step. I guess this saves me from having to make a union of the map :)

Reply:
Ah, that makes sense. I think perhaps then that extending SSLConfig to include both BackendSet and Listener SSL configuration would likely result in cleaner code. Thoughts?

Reply:
Yeah, you're right, Or maybe have a separate struct to hold both